src/components/Details/TranscriptSearch/TranscriptSearch.tsx

```diff
@@ -100,14 +101,97 @@ const TranscriptSearch: FC<TranscriptSearchProps> = ({
       if (!searchedTerm.trim()) {
         return sentences.data;
       }
-      const cleanedQuery = cleanText(searchedTerm);
+
+      let newTerm = searchedTerm;
+
+      // get beginning restricter
+      const beginIndStart = newTerm.indexOf('^"');
+      let beginRestriction = "";
+      if (beginIndStart !== -1) {
+        const beginIndFinal = newTerm.substring(beginIndStart + 2).indexOf('"') + beginIndStart + 2;
+        beginRestriction = newTerm.substring(beginIndStart + 2, beginIndFinal);
+        newTerm = newTerm.substring(0, beginIndStart) + newTerm.substring(beginIndFinal + 1);
+      }
+
+      // get ending restricter
+      const endIndStart = newTerm.indexOf('$"');
+      let endRestriction = "";
+      if (endIndStart !== -1) {
+        const endIndFinal = newTerm.substring(endIndStart + 2).indexOf('"') + endIndStart + 2;
+        endRestriction = newTerm.substring(endIndStart + 2, endIndFinal);
+        newTerm = newTerm.substring(0, endIndStart) + newTerm.substring(endIndFinal + 1);
+      }
+
+      // get all indices of double quotes
+      const regex = /"/gi,
+        indices = [],
+        required = [],
+        banned = [];
+      let result;
+      while ((result = regex.exec(newTerm))) {
+        indices.push(result.index);
+      }
+
+      // find everything inside these double quotes and remove them
+      for (let i = Math.floor(indices.length / 2) - 1; i >= 0; i--) {
+        if (newTerm.substring(indices[i * 2] - 1, indices[i * 2]) === "!") {
+          banned.push(newTerm.substring(indices[i * 2] + 1, indices[i * 2 + 1]));
+        } else {
+          required.push(newTerm.substring(indices[i * 2] + 1, indices[i * 2 + 1]));
+        }
+        newTerm =
+          newTerm.substring(0, indices[i * 2]) +
+          newTerm.substring(indices[i * 2 + 1] + 1, newTerm.length);
+      }
+
```

Comment:
can you pull everything from line 104-145ish into a utility function that just lives a neighboring folder? e.g. util/<function for getting restricter>, util/<function for getting double quotes>, etc?  Cleaner and more testable to keep the logic outside of views, as each function can be assessed on its sole purpose.

Reply:
Will do this once I fix John's bugs, thanks for the suggestion!